airbrake/notifier.py

```diff
@@ -186,15 +186,13 @@ def log(self, exc_info=None, message=None, filename=None,
                    'notifier': self.notifier,
                    'environment': environment,
                    'session': session}
-
-        return self.notify(payload)
+        return self.notify(json.dumps(payload, cls=utils.FailProofJSONEncoder))
```

Comment:
JSON encoding was not being covered in your tests (`notify` is mocked), so I moved the json encoding here, and `notify` now expects the serialized request body.